framework/src/utils/MooseUtils.C

```diff
@@ -145,6 +145,32 @@ parallelBarrierNotify(const Parallel::Communicator & comm)
   comm.barrier();
 }
 
+void serialBegin(const libMesh::Parallel::Communicator & comm)
+{
+  // unless we are the first processor...
+  if (comm.rank() > 0)
+  {
+    // ...wait for the previous processor to finish
+    int dummy = 0;
+    comm.receive(comm.rank() - 1, dummy);
+  }
+  else
+    mooseWarning("Entering serial execution block (use only for debugging)");
+}
+
+void serialEnd(const libMesh::Parallel::Communicator & comm)
+{
+  // unless we are the last processor...
+  if (comm.rank() + 1 < comm.size())
+  {
+    // ...notify the next processor of its turn
+    int dummy = 0;
+    comm.send(comm.rank() + 1, dummy);
+  }
+  else
+    mooseWarning("Leaving serial execution block (use only for debugging)");
```

Comment:
In `serialBegin()` you are printing the warning out on all processors but the first (except serial case).
In `serialEnd()` you are printing the warning out on all processors but the last (except serial case).

Messaging is different between the two and doesn't make sense if you were to save output streams to separate files. Yeah, I realize that it works correctly for the serial case. I'm just suggesting that you remove the `else` keyword.